src/azure-cli-core/azure/cli/core/_identity.py

```diff
@@ -0,0 +1,268 @@
+# --------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# --------------------------------------------------------------------------------------------
+
+import os
+import json
+
+from ._environment import get_config_dir
+from .util import get_file_json
+
+from knack.util import CLIError
+from knack.log import get_logger
+
+from azure.identity import (
+    AuthProfile,
+    InteractiveBrowserCredential,
+    DeviceCodeCredential,
+    UsernamePasswordCredential,
+    ClientSecretCredential,
+    CertificateCredential,
+    ManagedIdentityCredential
+)
+
+_CLIENT_ID = '04b07795-8ddb-461a-bbee-02f9e1bf7b46'
+logger = get_logger(__name__)
+
+_SERVICE_PRINCIPAL_ID = 'servicePrincipalId'
+_SERVICE_PRINCIPAL_TENANT = 'servicePrincipalTenant'
+_ACCESS_TOKEN = 'accessToken'
+_SERVICE_PRINCIPAL_CERT_FILE = 'certificateFile'
+_SERVICE_PRINCIPAL_CERT_THUMBPRINT = 'thumbprint'
+
+
+def _load_tokens_from_file(file_path):
+    if os.path.isfile(file_path):
+        try:
+            return get_file_json(file_path, throw_on_empty=False) or []
+        except (CLIError, ValueError) as ex:
+            raise CLIError("Failed to load token files. If you have a repro, please log an issue at "
+                           "https://github.com/Azure/azure-cli/issues. At the same time, you can clean "
+                           "up by running 'az account clear' and then 'az login'. (Inner Error: {})".format(ex))
+    return []
+
+
+def _delete_file(file_path):
+    try:
+        os.remove(file_path)
+    except FileNotFoundError:
+        pass
+
+
+class Identity:
+    """Class to interact with Azure Identity.
+    """
+    def __init__(self, authority, tenant_id):
```

Comment:
>authority [](start = 23, length = 9)

check authority cannot be None

Reply:
I prefer not to, because Azure Identity has a default value for `authority`:

https://github.com/chlowell/azure-sdk-for-python/blob/7be1ba7287a182778bcaa29f4dc1ba82ff949ea2/sdk/identity/azure-identity/azure/identity/_internal/msal_credentials.py#L92

```py
authority = kwargs.pop("authority", None) or get_default_authority()
```